driver-core/src/main/com/mongodb/internal/connection/CommandMessage.java

```diff
@@ -215,8 +216,14 @@ private boolean useOpMsg() {
         return getOpCode().equals(OpCode.OP_MSG);
     }
 
-    private List<BsonElement> getExtraElements(final SessionContext sessionContext) {
+    private List<BsonElement> getExtraElements(final OperationContext operationContext) {
+        SessionContext sessionContext = operationContext.getSessionContext();
+        TimeoutContext timeoutContext = operationContext.getTimeoutContext();
+
         List<BsonElement> extraElements = new ArrayList<>();
+        if (!getSettings().isCryptd()) {
+            timeoutContext.addTimeoutElements(extraElements);
```

Comment:
Something about extending extraElements in `TimeoutContext` feels strange.  Feels like the logic should be here.